packages/block-editor/src/components/block-popover/index.js

```diff
@@ -88,6 +124,47 @@ function BlockPopover(
 
 		return {
 			getBoundingClientRect() {
+				let postRootElement;
+				// The zoom out view has a vertical block toolbar that should always
+				// be on the edge of the canvas, aligned to the top of the currently
+				// selected section. This condition changes the anchor of the toolbar
+				// to the section instead of the block to handle blocks that are
+				// not full width and nested blocks to keep section height.
+				if ( isZoomOut && isSectionSelected ) {
+					// if the rootSectionElement is undefined then we need to recurse up the DOM tree
+					// to find the element with ` is-root-container` classname
+					// FIXME: we should not rely on classnames to find the
+					// root section element
```

Comment:
```suggestion
					// If the rootSectionElement is undefined then recurse up the DOM tree
					// to find the element with ` is-root-container` classname.
					// This can then be used as the anchor point.
					// FIXME: we should not rely on classnames to find the
					// root section element
```

Nit: but I think it improves it further.